### ISSUE TITLE
StackSet shouldn't become ready if it's broken

### ISSUE THREAD

**Issue by linki** [zalando-incubator/stackset-controller#47]
Broken StackSets are sometimes considered ready. Therefore traffic will be switched to it and results in downtime.

/cc @dryewo

**Comment by dryewo** [comment]
It once accepted a stackset manifest that contained the following under `spec.stackTemplate.spec`:

```
service:
  ports:
  - name: ingress
    port: "80"
    protocol: TCP
    targetPort: "8080"
```

Port numbers are strings, which is not valid. It successfully created a couple of stack then.

Later, new stacksets were not processed, no new stacks were created.
@linki please paste some logs here.

Ideally, such validation errors should be detected very early and definitely not affect other stacksets.

**Comment by mikkeloscar** [comment]
Depends on #10 which depends on Kubernetes v1.11

**Comment by mikkeloscar** [comment]
Fixed in #52
